src/main/java/com/jvm_bloggers/core/rss/BlogPostsController.java

```diff
@@ -19,23 +23,53 @@
 import static com.google.common.base.MoreObjects.firstNonNull;
 
 @RestController
-@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class BlogPostsController {
+    public static final String RSS_FEED_MAPPING = "/pl/rss";
 
     private final AggregatedRssFeedProducer rssProducer;
+    private final RssToJsonConverter rssToJsonConverter;
+
+    private final Integer defaultLimit;
+
+    @Autowired
+    public BlogPostsController(AggregatedRssFeedProducer rssProducer,
+                               RssToJsonConverter rssToJsonConverter,
+                               @Value("${generated.rss.entries.limit}") Integer defaultLimit) {
+        this.rssProducer = rssProducer;
+        this.rssToJsonConverter = rssToJsonConverter;
+        this.defaultLimit = defaultLimit;
+    }
 
     @SneakyThrows
-    @RequestMapping("/pl/rss")
+    @RequestMapping(RSS_FEED_MAPPING)
     public void getRss(HttpServletRequest request, HttpServletResponse response,
-        PrintWriter writer, @RequestParam(required = false) Integer limit,
-        @Value("${generated.rss.entries.limit}") int defaultLimit,
-        @RequestParam(required = false) Set<String> excludedAuthors) {
-        limit = firstNonNull(limit, defaultLimit);
-        response.setContentType(MediaType.APPLICATION_ATOM_XML_VALUE);
-        new SyndFeedOutput().output(
-            rssProducer.getRss(request.getRequestURL().toString(), limit, excludedAuthors),
-            writer
-        );
+                       PrintWriter writer,
+                       @RequestParam(required = false) Integer limit,
+                       @RequestParam(required = false) Set<String> excludedAuthors,
+                       @RequestParam(required = false, defaultValue = "xml") String format) {
+        Integer checkedLimit = firstNonNull(limit, defaultLimit);
+
+        SyndFeed feed =
+            rssProducer.getRss(request.getRequestURL().toString(), checkedLimit, excludedAuthors);
+
+        if (SupportedFormat.XML.toString().equalsIgnoreCase(format)) {
```

Comment:
I'd also consider splitting this into to separate methods each with it's own `@RequestMapping` like:
`@RequestMapping(path=RSS_FEED_MAPPING, produces = "application/json")`